### ISSUE TITLE
Don't abandon AngularDart, don't give up on AngularDart

### ISSUE THREAD

**Issue by insinfo** [dart-lang/angular#1866]
Don't abandon AngularDart, don't give up on AngularDart, me and my company are investing heavily in AngularDart, we are migrating all our web applications to AngularDart, I love Flutter for Mobile, but Flutter is not WEB-productive. WEB, we can't use many existing HTML and CSS templates, because in Flutter we can't use HTML and CSS.

* [x] Dart SDK 2.5.2
* [x] AngularDart 5.3.0
* [x] Windows 1903
* [x] Chrome 77

**Comment by supermuka** [comment]
My company is investing heavily projects using AngularDart too. We are really fan this powerful framework and the possibility to use all `html` and `css` features necessary to complex web app.

_**Long life to the AngularDart and Components!**_

**Comment by oonid** [comment]
the documentation site still updated daily [^1],
but the framework has no more updates since August 2019. [^2][^3]

[^1]: https://github.com/dart-lang/site-angulardart/commits/master
[^2]: https://github.com/dart-lang/angular/commits/github-sync
[^3]: https://github.com/dart-lang/angular/commits/master

**Comment by oonid** [comment]
good news, the framework [^1] get updates :laughing: 

[^1]: https://github.com/dart-lang/angular/commits/master

**Comment by DevAlves1993** [comment]
> good news, the framework [^1](https://github.com/dart-lang/angular/commits/master) get updates laughing

Really good news, I was really worried about the inactivity of repo. I hope Flutter will take the developers to try AngularDart

**Comment by insinfo** [comment]
@DevAlves1993 
I totally agree with you
>  I hope Flutter will take the developers to try AngularDart

**Comment by insinfo** [comment]
I see AngularDart as a much cleaner, nicer, organized and simple version of Angular. I particularly see TypeScript syntax very ugly and dirty. There are some areas of AngularDart that can improve, especially the router part and simpler forms of serialization.

**Comment by GowthamF** [comment]
I started learning Angular with TypeScript using some Udemy tutorials but I didn't even understand anything. So I had to give up on Angular. But After Flutter announces Flutter for Web I started learning Dart. It is really easy to learn. Since Flutter for Web is not yet ready for production, I moved to AngularDart. I learned everything using Documents. I was not able to find any Video Tutorials. Documentation is perfect for a beginner to learn it. I would really need Angular Dart to be added with some more features without giving it up.

**Comment by Bellski** [comment]
**Long life to the AngularDart and Components!**

**Comment by Dimous** [comment]
**We need an official status update.**

I heard Google uses AngularDart for AdWords and other teams also prefer Dart version. Is it still true?

**Comment by Bellski** [comment]
> **We need an official status update.**
> 
> I heard Google uses AngularDart for AdWords and other teams also prefer Dart version. Is it still true?

Problem that AdWords was on GWT and we know what happened(

**Comment by quangv** [comment]
@Bellski 
> Problem that AdWords was on GWT and we know what happened

What happened?

**Comment by Bellski** [comment]
> @Bellski
> 
> > Problem that AdWords was on GWT and we know what happened
> 
> What happened?

Google moved from GWT to Dart. And stopped support GWT.

**Comment by insinfo** [comment]
@Bellski 

> Long life to the AngularDart and Components!

I really hope Google will not abandon AngularDart, otherwise my organization and I will be greatly damaged.
If Flutter is successfully ported to the Web.
I think AngularDart runs on Flutter, because the angular templates HTML can be compiled for Flutter widgets. I've seen some projects here on github that use compiled XML for Flutter widgets, so this is very possible.

**Comment by steida** [comment]
![50913172_10157063174634826_274670163670859776_o](https://user-images.githubusercontent.com/66249/70857931-20d87200-1ef8-11ea-9191-49c5a93bbefe.jpg)

**Comment by timsneath** [comment]
Hi there -- thank you for your passion for AngularDart! It's always exciting to see others who are getting value out of this project. It's encouraging to see this issue get a ton of thumbs up. 

Firstly, let me (partly) reassure you. AngularDart is not going away and is as actively developed as ever, although we _have_ shifted our focus a little. 

As you have noticed, our activity on GitHub has become more bursty in recent months. Because of the heavy usage of AngularDart internally within Google, we moved the primary source of truth to [google3](https://dl.acm.org/doi/pdf/10.1145/2854146?download=true), our internal mono-repo. Our goal is to push out new code to GitHub periodically, and we're in the process of automating this so that we’re able to improve the frequency of these syncs. And as you've seen, in the past few days, [a large number of changes have landed](https://github.com/dart-lang/angular/commits/master) from a broad set of contributors that bring GitHub up to date.

But yes: we also want to be honest that our focus has shifted somewhat, recognizing the heavy internal usage of AngularDart and comparatively low external developer base. AngularDart is ideal for large-scale data-heavy apps like AdWords with millions of lines of code, but there just aren't that many of this kind of app externally, and we've gradually recognized that. Externally we have a loyal but relatively small following, but AngularDart is probably never going to be a broad general-purpose web framework. Conversely, we have a growing audience of thousands of Google engineers using AngularDart internally, and so most of the features we build are with that audience in mind, knowing that what they need will likely also be of value to external developers.

Lastly, we want to recognize that with the emergence of Flutter’s support for the web, we have other motivations to continue to develop Dart web scenarios - from improved IDE tooling to compilers. In fact, our level of investment in web tooling for Dart is higher than ever, with more engineers working in this space than at any previous point in the project's history.

So, to summarize:

- We're continuing to develop AngularDart, although our orientation has slightly shifted towards our internal audience, where it's proven very successful.
- We will continue to release AngularDart builds externally to GitHub. Success for the project isn't measured by us in how many millions of external web developers adopt it, but we are delighted to have you as a customer if it suits your needs.

**Comment by steida** [comment]
Hahahahahahahahahahaha. Idiots.

**Comment by steida** [comment]
@timsneath You are such a liar. Just tell the poor Google dev victims they are doomed and fucked up by Google yet again. Those lies per lies, do you even care about your karma?

**Comment by steida** [comment]
@timsneath Please, don't lie.

**Comment by steida** [comment]
Anyone who believes it is an idiot.

**Comment by MitaiGit** [comment]
AngularDart the best fraemwork in the world and it can become mega popular for this you need to start writing normal documentation, which would be supplemented, what is this tutorial where there is no word about the output of images on the site? how do I make friends with DjangoRestFramework or PHP? there are not so many server languages, you can write how to make friends with them, we noobs also want to be part of the team ANGULARDART

**Comment by CoreyCole** [comment]
> AngularDart is ideal for large-scale data-heavy apps like AdWords with millions of lines of code, but there just aren't that many of this kind of app externally

Another use case for AngularDart is sharing the business logic layer from a flutter app with a web app. As flutter for web matures, this probably isn't a long-term use case, but flutter web maturity seems a long way out especially with plugin compatibilities.

**Comment by supermuka** [comment]
@timsneath, I think it is very important to have information about the next steps of Google products.

First, it is a gratitude that we have Google sharing its technologies. As a company admired for its audacity, business models and technology many seek to mirror and use what is available to it, especially when open source.

I have been following and using Dart Lang for several years. It is a wonderful technology. However, one point has always made me apprehensive and anxious about the future: Lack of clear communication about the next steps, especially about the scope of development and release milestones.

As an open source product, in which many use and would also like to contribute to development, I see as important an additional effort to clearly manage communication, thereby aligning the expectations of both Google's internal team and external community.

An improvement in this sense, in addition to aligning expectations, would cause the community in general to create greater confidence about the future and thus generate greater adoption of Dart Lang, AngularDart, etc.

The technology is fantastic, I think adding a dash of management would make it even better.

**Comment by michalpie** [comment]
@timsneath How about AngularDart  Components (angular_components) package?  Will you also continue to release it on GitHub?

**Comment by CoreyCole** [comment]
@michalpie see here https://github.com/dart-lang/angular_components/issues/356#issuecomment-553591102

It is the same story as AngularDart, slower release cadence pace

**Comment by jodinathan** [comment]
We have almost no interest to change from AngularDart to Flutter for web in the next years because we have so many components being used that would need a great amount of work to make the change.
However, if there were something that can export angular dart components to widgets then it is a whole another story...

**Comment by stanislasdrg** [comment]
I am simply amazed by how little effort was put by Google in pushing the Dart Version of Angular.
It is simply miles ahead of Angular Ts in terms of simplicity and intuitivity.
Working with AngularDart has been a real pleasure so far. It would only take better documentation and potentially some bridges to Flutter to make an easy alternative to Angular Ts.

**Comment by levinem** [comment]
Is the version of Angular targeted by AngularDart internally at Google keeping up with the latest regular Angular releases (i.e. the 9.0 version)?  The version on github seems to be a bit behind the main Angular releases.  I am almost more inclined to use the regular Angular only to take advantage of the new Ivy capabilities versus those in the 5.0 Angular version targeted by AngularDart.  In general, though, thanks so much for your work on this product!

**Comment by kevmoo** [comment]
> Is the version of Angular targeted by AngularDart internally at Google keeping up with the latest regular Angular releases (i.e. the 9.0 version)? 

Hey @levinem – while both flavors of Angular (JS/TS vs Dart) share a history, they have diverged in many ways – mostly to align better with features in their respective platforms and to address issues with their core customers.

**Comment by levinem** [comment]
> > Is the version of Angular targeted by AngularDart internally at Google keeping up with the latest regular Angular releases (i.e. the 9.0 version)?
> 
> Hey @levinem – while both flavors of Angular (JS/TS vs Dart) share a history, they have diverged in many ways – mostly to align better with features in their respective platforms and to address issues with their core customers.

Thanks @kevmoo , I had thought that AngularDart was a Dart interface to the Angular main compiler, but it sounds like the underlying compiler infrastructures are not fundamentally connected and that AngularDart will try to keep the main interfaces and concepts similar to those in the JS/TS Angular project to enable a familiar interface for existing JS/TS Angular developers.

**Comment by kevmoo** [comment]
> AngularDart will try to keep the main interfaces and concepts similar to those in the JS/TS Angular project to enable a familiar interface for existing JS/TS Angular developers.

@levinem – I wouldn't go that far. The engineering teams are friendly, but they are separate teams. There is no explicit goal to keep things aligned – but there is no incentive to diverge either, unless there is a good reason.

**Comment by Stargator** [comment]
@timsneath I worked for 2 years on a project that was angularjs, in the last 8 months I was on the project, the team migrated to AngularDart. We rebuilt everything in AngularDart.

We also created Dart-based command line tools to ease our development workflow.

We created a generator tool to build our domain classes based on whatever client we were developing for.

Lastly, we created our own version of Pub to host our internal code.

The team was sold on the migration due to A) Dart's capabilities, B) Ease of taking existing web elements (HTML, CSS, and JS) into AngularDart variations, and C) Google's own investment and use of AngularDart for its Adwords, and other internal products.

The project is within the government space, so I cannot share details. But the product we developed consisted of over 15-20 microservices.

I hope this can tell you that it is absolutely an enterprise application. We needed Dart's strong type system so that we reduce the ambiguity within our logic.

AngularDart has a space externally from Google, but Google HAS to be public with its support.

My last point, AngularDart and the Dart platform itself, should allow `dart:html` to exist so that other frameworks and/or smaller apps can be built.

If Google doesn't show support and love for Dart and AngularDart like it does for Flutter, then it's all going to fall apart. That would be a shame, because there's real gold here.

And I concur with @supermuka about next steps for Dart and Angular Dart. These need to be public, even if short-term. Otherwise, no one will engage with what appears to be a ghost town run by ghosts.

**Comment by insinfo** [comment]
@timsneath 
First of all, I would like to thank you for your words and feedback.

@Stargator  

> If Google doesn't show support and love for Dart and AngularDart like it does for Flutter, then it's all going to fall apart. That would be a shame, because there's real gold here.

I totally agree with you, I am also working on several AngularDart projects for the Government of Brazil, I love AngularDart, I love the dart, we cannot let this wonderful tool die.

**Comment by kevmoo** [comment]
We're actively working to onboard someone to start the external rolls of angulardart. We're blocked by some paperwork, but we're working through it.

Thank you all for your interest and patience!

**Comment by anton-marchenko** [comment]
> AngularDart is ideal for large-scale data-heavy apps like AdWords with millions of lines of code

@timsneath
Sorry about non-subject question. Please, could you tell me how you develop such large-scale apps if [there is no](https://github.com/dart-lang/angular_components/issues/142#issuecomment-378810790) even [data-tables](https://material.io/components/data-tables/) in angular_components? Do you develop any internal Google UI Kit for that?

Because now we are developing an Angular Dart application (it was customer's choice) and we have a lot of question about angular_components. Our UI/UX architect makes mock ups according to https://material.io/design/ but we can not found some material components (data-tables or grid system, for example) in angular_components for design implementation.

Could you recomend something for us? Thank you!

**Comment by oonid** [comment]
@anton-marchenko 
Google has its own data table components,
that's planned to release to angular_components [since Q4 2018](https://github.com/dart-lang/angular_components/issues/356#issue-382416769), but still not happening till today.

there are several open-source components that [you can use from pub.dev](https://pub.dev/dart/packages?q=datatables+components).

**Comment by mmcc007** [comment]
This video series is a great intro to AngularDart for complete beginners:
https://www.youtube.com/watch?v=9X88k6kTuNw&list=PL3UwowPBo8XHbU4llLYZKY-FOofDLMv8S
Works thru an app from scratch, all the way to a working app with a firebase backend.

**Comment by szepeshazi** [comment]
@anton-marchenko We have build a number of Material Components that were missing from the official Angular Components library (data table included). See the gallery at https://skawa-universe.github.io/skawa_components_example/, the sources https://github.com/skawa-universe/skawa_components, and the published packages at https://pub.dev/packages?q=skawa+components

**Comment by anton-marchenko** [comment]
> @anton-marchenko We have build a number of Material Components that were missing from the official Angular Components library (data table included). See the gallery at https://skawa-universe.github.io/skawa_components_example/, the sources https://github.com/skawa-universe/skawa_components, and the published packages at https://pub.dev/packages?q=skawa+components

@szepeshazi Thank you for your feedback.  Unfortunately we have a limit time budget for our project. Because of this we decided to rewrite the whole [our project to Angular TypeScript.](https://github.com/skawa-universe/skawa_components/issues/63#issuecomment-597628310) 
Moreover, we [did NOT found](https://github.com/skawa-universe/skawa_components/issues/63)  that skawa table is component that completely satisfying our humble needs.

**Comment by axegon** [comment]
> Hi there -- thank you for your passion for AngularDart! It's always exciting to see others who are getting value out of this project. It's encouraging to see this issue get a ton of thumbs up.
> 
> Firstly, let me (partly) reassure you. AngularDart is not going away and is as actively developed as ever, although we _have_ shifted our focus a little.
> 
> As you have noticed, our activity on GitHub has become more bursty in recent months. Because of the heavy usage of AngularDart internally within Google, we moved the primary source of truth to [google3](https://dl.acm.org/doi/pdf/10.1145/2854146?download=true), our internal mono-repo. Our goal is to push out new code to GitHub periodically, and we're in the process of automating this so that we’re able to improve the frequency of these syncs. And as you've seen, in the past few days, [a large number of changes have landed](https://github.com/dart-lang/angular/commits/master) from a broad set of contributors that bring GitHub up to date.
> 
> But yes: we also want to be honest that our focus has shifted somewhat, recognizing the heavy internal usage of AngularDart and comparatively low external developer base. AngularDart is ideal for large-scale data-heavy apps like AdWords with millions of lines of code, but there just aren't that many of this kind of app externally, and we've gradually recognized that. Externally we have a loyal but relatively small following, but AngularDart is probably never going to be a broad general-purpose web framework. Conversely, we have a growing audience of thousands of Google engineers using AngularDart internally, and so most of the features we build are with that audience in mind, knowing that what they need will likely also be of value to external developers.
> 
> Lastly, we want to recognize that with the emergence of Flutter’s support for the web, we have other motivations to continue to develop Dart web scenarios - from improved IDE tooling to compilers. In fact, our level of investment in web tooling for Dart is higher than ever, with more engineers working in this space than at any previous point in the project's history.
> 
> So, to summarize:
> 
> * We're continuing to develop AngularDart, although our orientation has slightly shifted towards our internal audience, where it's proven very successful.
> * We will continue to release AngularDart builds externally to GitHub. Success for the project isn't measured by us in how many millions of external web developers adopt it, but we are delighted to have you as a customer if it suits your needs.

So in essence, do you plan on making any sorts of releases anytime soon? This is a dead serious question since we have an extremely large project at work, which by all means is comparable to AdWords. We have reached a point where we need to re-do a lot of the frontend, and while I rarely do anything there, I would like to use **anything other** than javascript(or typescript, which still lives within the Stephen King realm of horror that is the npm ecosystem: fun to read, not so much to live in). With this in mind, AngularDart seems like the only viable alternative. New releases and updates, 

![PLEASE](https://www.meme-arsenal.com/memes/c9760e82bae80e5df7d8521896dec4bf.jpg)

**Comment by stanislasdrg** [comment]
Come on @axegon, soldier on. Microsoft just acquired npm and will finally make it super reliable.
/s
What do you currently miss from AngularDart apart from UI comps ? The framework is a bit rough on the edges, but surely nothing is a deal breaker ?

Personally I don't get the enthusiasm for Flutter web. Correct me if I am wrong but it will make apps run on a canvas, the same way Flash worked ? If so performance will never be amazing.

**Comment by axegon** [comment]
@stanislasdrg even if an alien species acquires npm, it is far beyond repair(assuming there was something to repair, considering it's js). To be fair I don't miss anything from AngularDart. But much like anything else, you don't want to use something that isn't supported in any way, do you? It's all fun and games until everything stops working.

Flutter web, while promising is not a viable solution for anything other than mobile web I'm afraid. I played around with it some time ago - great performance, 0 usability on desktop(you can check my github.io for an example).

**Comment by kevmoo** [comment]
We are actively working to bring on the resources needed to continue releasing this package.

Sadly, timing aligned with global events to slow everything down.

Please know this is important to the team and we continue to make progress – just at a pace we all wish was quicker.

**Comment by insinfo** [comment]
@kevmoo
First of all, I would like to thank you for your words and feedback.

**Comment by graphicbeacon** [comment]
@kevmoo @timsneath Thank you for reassuring us that you intend to keep the framework going. I was wondering whether to keep making AngularDart tutorials so your responses tell me to continue doing so. @mmcc007 Thanks for sharing the playlist 😉

**Comment by Overfacto** [comment]
The flutter is inspired by React. I know redux is pretty good pattern but it has limits on any web architectures, and not friendly web than Angular.

Cheer AngularDart, you should attract nest, ivy, karma or many developers. I hope really to activate this project.

**Comment by refi64** [comment]
Looks like we're getting another sync soon! https://github.com/dart-lang/angular/pull/1892

**Comment by jodinathan** [comment]
> Looks like we're getting another sync soon! #1892

that pr seems quite big

**Comment by e-belair** [comment]
`sdk: ">=2.8.1 <3.0.0"`, that should solve many dependencies constraints errors

**Comment by Stargator** [comment]
@kevmoo @timsneath I recommend you pin this issue so folks find it easier and hopefully avoid creating new issues on the same topic.

**Comment by insinfo** [comment]
Very good to know that there is work behind the scenes.

https://github.com/dart-lang/angular/pull/1892

**Comment by francovp** [comment]
I think [this tweet](https://twitter.com/timsneath/status/1285415204059144198) from the Group Product Manager for Flutter and Dart makes it clear

![image](https://user-images.githubusercontent.com/2883350/88011430-4c9e8480-cae5-11ea-89b0-2bf4f4b1cf16.png)

**Comment by graphicbeacon** [comment]
> I think [this tweet](https://twitter.com/timsneath/status/1285415204059144198) from the "Group Product Manager for Flutter and Dart" from Google makes it clear.
> 
> Flutter Web is the priority for Google
> 
> ![image](https://user-images.githubusercontent.com/2883350/88011430-4c9e8480-cae5-11ea-89b0-2bf4f4b1cf16.png)

Guess I'll be cutting down on the AngularDart tutorials then.

**Comment by DartViet** [comment]
> I think [this tweet](https://twitter.com/timsneath/status/1285415204059144198) from @timsneath "Group Product Manager for Flutter and Dart" from Google makes it clear.
> 
> Flutter Web is the priority for Google
> 
> ![image](https://user-images.githubusercontent.com/2883350/88011430-4c9e8480-cae5-11ea-89b0-2bf4f4b1cf16.png)

This is too bad... you don't have to draw everything on canvas. Angular Dart has so much potential... . Dart is getting fame because of Flutter, people start learning it,... Angular Dart is good for anyone familiar with Angular TS, and this is the right tool to use. Imagine... Angular Dart, Aqueduct, Flutter... we are kicking Javascript left and right... (as someone with background of .net C#, words can't describe how much I dislike Javascript) Come on.

**Comment by axegon** [comment]
> > I think [this tweet](https://twitter.com/timsneath/status/1285415204059144198) from @timsneath "Group Product Manager for Flutter and Dart" from Google makes it clear.
> > Flutter Web is the priority for Google
> > ![image](https://user-images.githubusercontent.com/2883350/88011430-4c9e8480-cae5-11ea-89b0-2bf4f4b1cf16.png)
> 
> This is too bad... you don't have to draw everything on canvas. Angular Dart has so much potential... . Dart is getting fame because of Flutter, people start learning it,... Angular Dart is good for anyone familiar with Angular TS, and this is the right tool to use. Imagine... Angular Dart, Aqueduct, Flutter... we are kicking Javascript left and right... (as someone with background of .net C#, words can't describe how much I dislike Javascript) Come on.

Absolutely! Granted that the sync goes through sometime in the near future, I think someone should fork it and try to build a community around it and hopefully keep it alive. Personally, I'd be more than happy to contribute, despite having some time constraints.

**Comment by supermuka** [comment]
> > > I think [this tweet](https://twitter.com/timsneath/status/1285415204059144198) from @timsneath "Group Product Manager for Flutter and Dart" from Google makes it clear.
> > > Flutter Web is the priority for Google
> > > ![image](https://user-images.githubusercontent.com/2883350/88011430-4c9e8480-cae5-11ea-89b0-2bf4f4b1cf16.png)
> > 
> > 
> > This is too bad... you don't have to draw everything on canvas. Angular Dart has so much potential... . Dart is getting fame because of Flutter, people start learning it,... Angular Dart is good for anyone familiar with Angular TS, and this is the right tool to use. Imagine... Angular Dart, Aqueduct, Flutter... we are kicking Javascript left and right... (as someone with background of .net C#, words can't describe how much I dislike Javascript) Come on.
> 
> Absolutely! Granted that the sync goes through sometime in the near future, I think someone should fork it and try to build a community around it and hopefully keep it alive. Personally, I'd be more than happy to contribute, despite having some time constraints.

Sometimes, I think, because AngularDart is **so **fantastic**** and Google uses it a lot internally, _Google may not want to share with the community this powerful structure_ that allows to develop in a pleasant way, magnificent, robust and complex applications.

I've been hoping to see updates for a long time, but it's becoming increasingly clear in the leaders' messages that the focus is different. Unfortunately!

Perhaps there will be two ways to see the evolution of AngularDart: working at Google :) or the community fork and evolve the framework itself.

**Comment by timsneath** [comment]
You might want to stop copying that image into every reply, since it makes it harder to follow the thread :)

I know that this is a nuanced message, which therefore runs the risk of misinterpretation, so let me reiterate a few things:
1. We're continuing to actively develop AngularDart
2. Our primary audience is more Google-centric than it used to be, because we're seeing heavy usage in critical projects that pay our salaries
3. We're continuing to ship new builds externally, and we have one on the way when we've resolved various sync issues
4. We know AngularDart is fantastic: and we're glad you know, @supermuka :)
5. Flutter's web support isn't ready yet, we know. We're investing there because we are excited about it's long-term potential and we want to give it the very best shot at success.
6. No need for the community to fork AngularDart, since we're continuing to release new builds -- although of course you're welcome to do so as you always have been -- it's open source, after all!

**Comment by oonid** [comment]
so,
have you checked the [new merge on master](https://github.com/dart-lang/angular/pull/1898) branch? :+1: 
[angular 6.0.0-dev](https://github.com/dart-lang/angular/commit/11429ea5a8ce57d0fad95a0a48089d5dedba407b)
did anyone try it?

**Comment by bshlomo** [comment]
@timsneath can you share with us a road map for flutter web support, it is crucial information that I think Google must share with the community.

**Comment by timsneath** [comment]
We're pretty open about our roadmap and progress, @bshlomo. Most recently, we posted this: 
https://medium.com/flutter/going-deeper-with-flutters-web-support-66d7ad95eb52

Our focus right now is getting web support into the Flutter `stable` channel, which means completing work on CanvasKit support, accessibility, performance and plugins.  That will give us a solid foundation for further feature development work.

**Comment by MitaiGit** [comment]
angulardart for creating sites (full-fledged sites) is the best choice, flutter now has a number of disadvantages that not everyone is ready to accept, as an example, not being able to copy text

**Comment by bshlomo** [comment]
Thank you @timsneath for the link abut flutter-web-support, very professional, and it seems that Google is doing a wonderful job.
Still, the unavoidable question is when?
We are in a situation that we need to make a decision with which technology to port & enhance a huge application that needs to be run in a browser & mobile from Java to Dart the alternatives are dart:html package, AngularDart or to wait for Flutter on the web 
Road map information will help us to make the right decision. 

No one can ignore Google's investment in Flutter, and it is pretty clear where the efforts are, and what will be in the future, but it will preferable if Google will be more clear about when we can start using Flutter on the web, even some rouge estimation will be most valuable to a lot of teams out there

**Comment by jimmyff** [comment]
@bshlomo as someone that's built & launched app on both Angular Dart & Flutter I would strongly recommend flutter for any new project. I can elaborate if you need but I don't want to derail this topic. Gitter is probably a better place to discuss, feel free to ping me.

**Comment by Sunbreak** [comment]
> @bshlomo as someone that's built & launched app on both Angular Dart & Flutter I would strongly recommend flutter for any new project. I can elaborate if you need but I don't want to derail this topic. Gitter is probably a better place to discuss, feel free to ping me.

Could we have a `Discussions` tab as to https://github.com/dear-github/dear-github/issues/44?

**Comment by jodinathan** [comment]
> @bshlomo as someone that's built & launched app on both Angular Dart & Flutter I would strongly recommend flutter for any new project. I can elaborate if you need but I don't want to derail this topic. Gitter is probably a better place to discuss, feel free to ping me.

You could create an article in medium or some other blog system.
We are very interested in your story and arguments to use flutter web for new projects.
In a few months ahead we will be in the same position as @bshlomo and this would help a lot in the decision.

**Comment by bshlomo** [comment]
Thank you for your interest in this subject
I am sure that this issue (Web/Desktop development with Dart) is a "must clear" and the responsibility is on Google's.
Google must share with us more information about Flutter and or Angular Dart directions, milestones, dates, etc..

You can't pull thousands of developers after you, while they are asking (and not getting) some relevant information that is a standard in this industry and we call it "product's road map" 

The effort, difficulty, challenge, and risk (and we all appreciate the wonderful job Google is doing on this) can't cause Google to ignore our needs as users for valuable information that will help us make our own decisions, I am sure that Google doesn't want that we lose money and time by making wrong decisions that result from a lack of basic but committed information.

Why send someone to start development with a dead product?
A development cycle can take years, why let us start a project with the wrong tools.
We don't want to guess, and this is not a quiz, we truly need answers and commitment, I sure that Google can supply them both.

About @jodinathan suggestion - article in medium - a very good idea
About @Sunbreak  suggestion - I don't know how to do it
About @jimmyff offer - I don't have Gitter, I will try to install it and ping you

**Comment by Sunbreak** [comment]
@bshlomo Thumbs up https://github.com/dear-github/dear-github/issues/44 may help a lot

And it seems that GitHub prefers frontend frameworks/libs than any than repos:

- https://github.com/zeit/next.js/discussions
- https://github.com/cuelang/cue/discussions
- https://github.com/cuelang/cue/discussions
- https://github.com/pikapkg/snowpack/discussion

**Comment by insinfo** [comment]
I use AngularDart extensively in a multitude of projects, in the company where I work, and particularly I love AngularDart.
I also use Flutter a lot for mobile applications and I love that too.
I think that Flutter has potential on the WEB for some scenarios, but not for all scenarios. I think that a WEB tool / framework for all scenarios has to support HTML and CSS and that AngularDart does very well.
I see that in the long term in the future if Flutter was executed in WebAssembly and using WebGPU this would bring high performance that would allow to port the html template language from Angular to Flutter this would bring the possibility of using Flutter with and without HTML, it already has some projects on GitHub that show that it is very possible to do this, for example there are some projects that show HTML rendering for Flutter widgets that would bring the best of both worlds.

**Comment by MitaiGit** [comment]
Please add tutorial, Angulardart + gRPC and shelf or aqueduct(dart for backend) mmmm?

**Comment by p554157atch** [comment]
As far as I know, we need to route GRPC request through a proxy, and certain things are still not supported for example streamed input and streamed output.

The grpc-web example which I already looked through requires a GRPC web capable server, and currently the only working implementation of such server is written in node.js I was unable to find an example that makes us of Golang server.

GRPC must be implemented as a protocol in the web browser similar to HTTP to make it work without hacks.

**Comment by refi64** [comment]
You can use [grpcwebproxy](https://github.com/improbable-eng/grpc-web/tree/master/go/grpcwebproxy) or Envoy for this which [I've used in a pure Dart stack before](http://github.com/refi64/matrix-invite-panel). That being said, this is getting rather off-topic for this issue and further discussion related to GRPC in particular should probably be elsewhere...

**Comment by timsneath** [comment]
Closing issue to avoid it becoming a suggestion box for all new feature requests :) 

We have a new release coming imminently; our [commitment to AngularDart is solid](https://github.com/dart-lang/angular/issues/1866#issuecomment-571851626), albeit with a more pronounced focus on Google's large scale apps like Google Ads and Play Console.

**Comment by kevmoo** [comment]
Please see https://groups.google.com/a/dartlang.org/g/announce/c/Kz84KNBcf3U !
